distributed/scheduler.py

```diff
@@ -1436,15 +1459,15 @@ def __init__(
         self.resource_restrictions = None
         self.loose_restrictions = False
         self.actor = False
-        self.prefix = None  # type: ignore
         self.type = None  # type: ignore
-        self.group_key = key_split_group(key)
-        self.group = None  # type: ignore
         self.metadata = None
         self.annotations = None
         self.erred_on = None
         self._rootish = None
         self.run_id = None
+        self.group = group
+        group.add(self)
+        group.prefix.add(self)
```

Comment:
Intuitively, I would expect `group.add(self)` to be sufficient and not require the caller to also do `group.prefix.add(self)`.

Why can't `TaskGroup.add` be extended to call `TaskPrefix.add`?

Same question for `transition` below

Reply:
Fair point, this is now aligned.